budgetr-backend/src/main/java/pl/mantiscrab/budgetr/auth/SecurityContextSignedInUsernameGetter.java

```diff
@@ -0,0 +1,21 @@
+package pl.mantiscrab.budgetr.auth;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import pl.mantiscrab.budgetr.domain.user.exceptions.BudgetrUserNotAuthenticatedException;
+import pl.mantiscrab.budgetr.domain.user.infrastructure.SignedInUsernameGetter;
+
+class SecurityContextSignedInUsernameGetter implements SignedInUsernameGetter {
```

Comment:
Getter? XD

Reply:
Seems reasonable to me... I can suggest "Provider" if you think it fits better

Reply:
You also doing in here authentication check so it's not getter.

Getter suggest object being anemic

Reply:
Ok, sounds right, so what would you suggest? Is "Provider" better, because it's not suggesting being anemic?

Reply:
yep is much better  :-)